index.js

```diff
@@ -204,6 +214,134 @@ function decorateCollection(collection) {
       }
     }
   };
+
+  // ensureIndex is deprecated but createIndex has exactly the
+  // same behavior
+
+  newCollection.ensureIndex = function(fieldOrSpec, options, callback) {
+    return newCollection.createIndex.apply(newCollection, Array.prototype.slice.call(arguments));
+  };
+
+  newCollection.insert = function(docs, options, callback) {
+    // Use undeprecated equivalents for the two use cases
+    if (Array.isArray(docs)) {
+      return newCollection.insertMany.apply(newCollection, Array.prototype.slice.call(arguments));
+    } else {
+      return newCollection.insertOne.apply(newCollection, Array.prototype.slice.call(arguments));
+    }
+  };
+
+  newCollection.remove = function(selector, options, callback) {
+    // Undeprecated equivalents
+    if (options && ((typeof options) === 'object') && options.single) {
+      arguments[1] = omit(arguments[1], [ 'single' ]);
+      return newCollection.deleteOne.apply(newCollection, Array.prototype.slice.call(arguments));
+    } else {
+      return newCollection.deleteMany.apply(newCollection, Array.prototype.slice.call(arguments));
+    }
+  };
+
+  newCollection.update = function(selector, doc, _options, callback) {
```

Comment:
`changes` or `updates` seems clearer instead of `doc` since it might not be the entire doc being replaced..

Reply:
Even if it's a total replace, `docUpdate` might be clearer without confusing again with `update` (as the mongodb docs do a bit).